core/src/main/scala/codesearch/core/index/LanguageIndex.scala

```diff
@@ -141,21 +153,16 @@ trait LanguageIndex[VTable <: DefaultTable] { self: DefaultDB[VTable] =>
 
   private def createCSearchResult(fullPath: String, lineNumber: Int): Option[CSearchResult] = {
     val relativePath = Path(fullPath).relativeTo(pwd).toString
-    relativePath.split('/').drop(2).toList match {
-      case libName :: version :: path =>
-        val decodedName       = URLDecoder.decode(libName, "UTF-8")
-        val (firstLine, rows) = Helper.extractRows(relativePath, lineNumber)
-        Some(
-          CSearchResult(s"$decodedName-$version",
-                        buildRepUrl(decodedName, version),
-                        CodeSnippet(
-                          path.mkString("/"),
-                          firstLine,
-                          lineNumber - 1,
-                          rows
-                        )))
-      case _ =>
-        None
+    packageName(relativePath).map { p =>
+      val (firstLine, rows) = Helper.extractRows(relativePath, lineNumber)
+      CSearchResult(p,
+                    CodeSnippet(
+                      relativePath.split('/').drop(4).mkString("/"),
```

Comment:
Can you show in comments how these paths look like? It's hard to follow `drop(4)`

Reply:
ok